# Defining the role


In [14]:
from anthropic import Anthropic
from dotenv import load_dotenv
import os

load_dotenv()
api_key = os.getenv("ANTHROPIC_API_KEY")
client = Anthropic(api_key=api_key)
model = "claude-haiku-4-5-20251001"

patient_data = """
P001: Female,45,BMI31.2,BP145/92,Glucose128,Chol240,NonSmoker,Exercise1
P002: Male,29,BMI24.1,BP118/76,Glucose92,Chol180,Smoker,Exercise2
P003: Female,62,BMI28.5,BP158/96,Glucose145,Chol260,NonSmoker,Exercise0
"""

prompt = f"""Analyze this patient data and identify health risks:
{patient_data}"""

system = "Analyze the patient dataset and identify common health patterns, key risk factors, and regional health concerns. Summarize the main findings and areas needing attention.."

messages = [{"role": "user", "content": prompt}]

response = client.messages.create(
    model=model,
    max_tokens=1500,
    messages=messages,
    system=system  
)

print(response.content[0].text)

# Patient Dataset Health Analysis

## Individual Risk Profiles

| Patient | Age | Gender | Key Risk Factors | Risk Level |
|---------|-----|--------|------------------|-----------|
| **P001** | 45 | F | Obesity (BMI 31.2), Hypertension (145/92), Elevated Glucose (128), High Cholesterol (240), Low Activity (1x/week) | **HIGH** |
| **P002** | 29 | M | Active Smoking, Borderline Cholesterol (180) | **MODERATE** |
| **P003** | 62 | F | Age 62+, Hypertension (158/96), Elevated Glucose (145), High Cholesterol (260), **Sedentary (0 exercise)** | **CRITICAL** |

---

## Key Health Patterns Identified

### 🔴 Critical Concerns

1. **Cardiovascular Disease Risk**: All three patients show hypertension; P003 requires urgent intervention
   - P001: Stage 2 (145/92)
   - P003: Stage 2 (158/96) + diabetes markers

2. **Metabolic Dysfunction Cluster**:
   - Elevated fasting glucose in P001 (128) and P003 (145) → prediabetes/diabetes risk
   - High cholesterol affecting 100% of cohort (range 180-260)

3

### V1 : You are a healthcare data analyst. Identify health risks in the data.


 Healthcare Data Analysis - Health Risk Assessment

 Individual Risk Profiles

 **P001: Female, 45 years old**
**Risk Level: MODERATE-HIGH**

| Risk Factor | Value | Status |
|-------------|-------|--------|
| BMI | 31.2 | ⚠️ **Obese** (>30) |
| Blood Pressure | 145/92 | ⚠️ **Stage 2 Hypertension** |
| Glucose | 128 | ⚠️ **Elevated** (prediabetic range) |
| Cholesterol | 240 | ⚠️ **High** (>200) |
| Exercise | 1 day/week | ⚠️ **Insufficient** |

**Key Concerns:** Metabolic syndrome cluster (obesity + hypertension + glucose dysregulation + hyperlipidemia)

---

 **P002: Male, 29 years old**
**Risk Level: LOW-MODERATE**

| Risk Factor | Value | Status |
|-------------|-------|--------|
| BMI | 24.1 | ✓ Normal |
| Blood Pressure | 118/76 | ✓ Normal |
| Glucose | 92 | ✓ Normal |
| Cholesterol | 180 | ✓ Desirable |
| Smoking | Yes | ⚠️ **MAJOR RISK** |
| Exercise | 2 days/week | ✓ Adequate |

**Key Concerns:** Smoking significantly increases cardiovascular and cancer risk despite otherwise favorable metrics

---

 **P003: Female, 62 years old**
**Risk Level: HIGH**

| Risk Factor | Value | Status |
|-------------|-------|--------|
| Age | 62 | ⚠️ **Increased CVD risk** |
| BMI | 28.5 | ⚠️ **Overweight** |
| Blood Pressure | 158/96 | 🔴 **Stage 2 Hypertension** (highest) |
| Glucose | 145 | 🔴 **Diabetic range** |
| Cholesterol | 260 | 🔴 **Significantly elevated** |
| Exercise | 0 days/week | 🔴 **Sedentary** |

**Key Concerns:** Multiple uncontrolled cardiovascular risk factors + sedentary lifestyle = **HIGH acute event risk**

---

 Priority Recommendations

| Patient | Priority Actions |
|---------|-----------------|
| **P001** | Lifestyle modification (diet/exercise), medication review for HTN/cholesterol, weight loss program |
| **P002** | **SMOKING CESSATION** (most urgent), maintain current healthy lifestyle |
| **P003** | **URGENT:** Medical intervention for HTN/diabetes control, cardiology referral, supervised exercise program |

### V2:Analyze the patient data and identify key health risks for each patient in simple words.

Patient Health Risk Analysis

**P001 - Female, 45 years old**
**Key Health Risks:**
- **Overweight** (BMI 31.2) - classified as obese
- **High blood pressure** (145/92) - Stage 2 hypertension
- **High blood sugar** (128) - prediabetic range
- **High cholesterol** (240) - increased heart disease risk
- **Low physical activity** (1 day/week)

**Main Concern:** High risk for heart disease and diabetes

---

 **P002 - Male, 29 years old**
**Key Health Risks:**
- **Smoking** - major health threat
- All other measurements are healthy (normal weight, BP, glucose, cholesterol)

**Main Concern:** Smoking is the primary risk; needs to quit smoking immediately

---

 **P003 - Female, 62 years old**
**Key Health Risks:**
- **Very high blood pressure** (158/96) - Stage 2 hypertension
- **High blood sugar** (145) - likely diabetic
- **Very high cholesterol** (260) - serious heart disease risk
- **Overweight** (BMI 28.5)
- **No exercise** (0 days/week) - sedentary lifestyle
- **Age 62** - increased age-related health risks

**Main Concern:** **HIGHEST RISK PATIENT** - Multiple serious conditions requiring urgent medical attention

---

**Summary:**
- **P003** = Most urgent intervention needed
- **P001** = Moderate-to-high risk, needs lifestyle changes
- **P002** = Young and mostly healthy, but smoking is critical issue

### v3:You are health care data expert.Analyse the given data answer the following patient wise 1) Risks 2)Patterns in thier health 3)Yes or no for hospital visit

 Patient Health Analysis

**P001 - Female, 45 years**

**Risks:**
- Obesity (BMI 31.2 - Class I)
- Hypertension Stage 2 (BP 145/92)
- Elevated cholesterol (240 mg/dL)
- Elevated fasting glucose (128 mg/dL) - prediabetic range
- Sedentary lifestyle (Exercise 1x/week)

**Health Patterns:**
- Metabolic syndrome indicators present
- Cardiovascular disease risk factors clustering
- Weight management issues affecting multiple biomarkers

**Hospital Visit:** **YES** ⚠️
*Recommend: Cardiology evaluation, metabolic workup, lifestyle intervention program*

---

 **P002 - Male, 29 years**

**Risks:**
- Active smoking (primary concern)
- Borderline overweight (BMI 24.1)
- All other metrics within normal ranges

**Health Patterns:**
- Generally healthy profile despite smoking
- Good exercise habits (2x/week) compensating somewhat
- No current metabolic dysfunction

**Hospital Visit:** **NO** ✓
*Recommend: Smoking cessation counseling, routine check-up in 6-12 months*

---

 **P003 - Female, 62 years**

**Risks:**
- **HIGHEST RISK PROFILE**
- Hypertension Stage 2 (BP 158/96)
- Elevated glucose (145 mg/dL) - diabetic range
- High cholesterol (260 mg/dL)
- Overweight (BMI 28.5)
- Sedentary (No exercise)
- Advanced age (62) + multiple comorbidities

**Health Patterns:**
- Uncontrolled cardiovascular & metabolic disease
- Cumulative risk for stroke/MI/kidney disease
- Rapid health deterioration potential

**Hospital Visit:** **YES - URGENT** 🔴
*Recommend: Immediate comprehensive evaluation, diabetes screening, ECG, blood pressure management, exercise prescription*

### v4:Analyze the patient dataset and summarize the main health patterns you observe. Identify any common risk factors, highlight the patients who may need the most attention, and explain the key health concerns shown in the data. Provide a brief summary of your findings.

 Patient Dataset Analysis Summary

 Main Health Patterns Observed

Overall Risk Profile
This small cohort shows **significant cardiovascular and metabolic health concerns**, particularly among older and higher BMI patients.

---

 Key Health Findings by Risk Area

 **1. Hypertension (Blood Pressure)**
- **P001 & P003**: Both have elevated BP (145/92 and 158/96) — **Stage 2 Hypertension**
- **P002**: Normal BP (118/76) — optimal
- **Risk Impact**: 67% of patients require BP management

**2. Metabolic Dysfunction**
| Patient | BMI | Glucose | Cholesterol | Status |
|---------|-----|---------|-------------|--------|
| P001 | 31.2 | 128 | 240 | Obese + Prediabetic + High Chol |
| P002 | 24.1 | 92 | 180 | Healthy profile |
| P003 | 28.5 | 145 | 260 | Overweight + Diabetic range + Very High Chol |

 **3. Lifestyle Factors**
- **Smoking**: P002 (smoker) adds cardiovascular risk despite healthy vitals
- **Exercise**: P003 (zero exercise) is most sedentary; P002 highest activity
- **BMI**: P001 and P003 overweight/obese (contributing factor to conditions)

---

Patients Requiring Most Attention

 🔴 **HIGHEST PRIORITY: P003 (62F)**
- Elevated BP (158/96)
- Glucose at 145 (diabetic range)
- Highest cholesterol (260)
- **Zero exercise** — completely sedentary
- **Age 62** adds compounding cardiovascular risk
- **Recommendation**: Urgent diabetes screening, hypertension management, cardiology referral

 🟡 **MODERATE PRIORITY: P001 (45F)**
- Obese (BMI 31.2)
- Elevated BP and glucose (prediabetic)
- High cholesterol
- **Recommendation**: Lifestyle modification (weight loss, exercise), medication evaluation

 🟢 **LOW PRIORITY: P002 (29M)**
- Optimal vitals and BMI
- **Risk**: Smoking habit undermines otherwise healthy profile
- **Recommendation**: Smoking cessation counseling

---

Key Health Concerns Summary

1. **Cardiovascular Risk** — 67% with hypertension; 33% with significantly elevated cholesterol
2. **Metabolic Syndrome Signs** — Multiple patients showing obesity + glucose + cholesterol abnormalities
3. **Lifestyle Gaps** — Inadequate exercise in 67% of cohort; one active smoker
4. **Age Factor** — P003's age combined with multiple risk factors creates compounding disease risk

**Overall Assessment**: This cohort needs aggressive preventive intervention, particularly P003 for potential diabetes management and cardiovascular disease prevention.

### v5: Analyze the patient dataset and identify common health patterns, key risk factors, and regional health concerns. Summarize the main findings and areas needing attention.

 Patient Dataset Health Analysis

 Individual Risk Profiles

| Patient | Age | Gender | Key Risk Factors | Risk Level |
|---------|-----|--------|------------------|-----------|
| **P001** | 45 | F | Obesity (BMI 31.2), Hypertension (145/92), Elevated Glucose (128), High Cholesterol (240), Low Activity (1x/week) | **HIGH** |
| **P002** | 29 | M | Active Smoking, Borderline Cholesterol (180) | **MODERATE** |
| **P003** | 62 | F | Age 62+, Hypertension (158/96), Elevated Glucose (145), High Cholesterol (260), **Sedentary (0 exercise)** | **CRITICAL** |

---

 Key Health Patterns Identified

 🔴 Critical Concerns

1. **Cardiovascular Disease Risk**: All three patients show hypertension; P003 requires urgent intervention
   - P001: Stage 2 (145/92)
   - P003: Stage 2 (158/96) + diabetes markers

2. **Metabolic Dysfunction Cluster**:
   - Elevated fasting glucose in P001 (128) and P003 (145) → prediabetes/diabetes risk
   - High cholesterol affecting 100% of cohort (range 180-260)

3. **Sedentary Lifestyle Crisis**: P003 has zero exercise with multiple comorbidities

---

Areas Requiring Immediate Attention

| Issue | Patients | Recommended Action |
|-------|----------|-------------------|
| **Hypertension Management** | P001, P003 | Urgent BP monitoring; medication review |
| **Metabolic Screening** | P001, P003 | HbA1c testing; diabetes prevention program |
| **Obesity Intervention** | P001 | Nutrition counseling; weight management plan |
| **Smoking Cessation** | P002 | Counseling; pharmacotherapy options |
| **Physical Activity** | P001, P003 | Structured exercise programs (target: ≥150 min/week) |

---

Recommended Interventions (Priority Order)

1. **P003** (Highest Priority): Comprehensive geriatric assessment, intensive lifestyle modification, BP/glucose management
2. **P001**: Weight loss program (target -5-10%), exercise progression, lipid medication review
3. **P002**: Smoking cessation support despite younger age and better overall metrics

# Healthcare Analysis Prompt Comparison

This notebook compares multiple system prompt versions for analyzing a small patient dataset with BMI, blood pressure, glucose, cholesterol, smoking status, and exercise frequency.

## Dataset

The analysis uses three patient records:

- P001: Female, 45, BMI 31.2, BP 145/92, Glucose 128, Cholesterol 240, Non-smoker, Exercise 1
- P002: Male, 29, BMI 24.1, BP 118/76, Glucose 92, Cholesterol 180, Smoker, Exercise 2
- P003: Female, 62, BMI 28.5, BP 158/96, Glucose 145, Cholesterol 260, Non-smoker, Exercise 0

## Prompt Version Comparison

| Version | Prompt Focus | Result Quality | Key Notes |
|---|---|---|---|
| V1 | Identify health risks in the data | Good | Strong patient-wise risk breakdown with clear tables and priority recommendations, but less focused on overall dataset patterns. |
| V2 | Explain key health risks in simple words | Good for beginners | Very readable and easy to understand, but less detailed and less structured for a formal healthcare analysis. |
| V3 | Patient-wise risks, health patterns, and hospital visit yes/no | Useful but too clinical | Includes clear action decisions, but the hospital visit labels may be too strong without more clinical context. |
| V4 | Summarize patterns, common risk factors, high-priority patients, and key concerns | Best | Best balance of clarity, structure, patient prioritization, common patterns, and brief summary. It directly matches the analysis goal. |
| V5 | Common patterns, risk factors, regional health concerns, areas needing attention | Mixed | Has a strong table and recommendations, but includes an irrelevant "regional health concerns" requirement and a factual issue about hypertension across all patients. |

## Best Performing Prompt

**V4 gave the best result.**

V4 worked best because it asked for:

- Main health patterns
- Common risk factors
- Patients needing the most attention
- Key health concerns
- A brief final summary

This produced the most complete and balanced output. It identified P003 as the highest-priority patient, P001 as moderate priority, and P002 as lower priority with smoking as the main concern.

## Final Recommendation

Use V4 as the preferred system prompt style for this healthcare analysis task:

```text
Analyze the patient dataset and summarize the main health patterns you observe. Identify any common risk factors, highlight the patients who may need the most attention, and explain the key health concerns shown in the data. Provide a brief summary of your findings.
```

This prompt is specific enough to guide the model while still leaving room for a useful, structured analysis.
